spec/unit/ec2_server_create_spec.rb

```diff
@@ -2113,5 +2112,39 @@
         end
       end
     end
+
+    describe "--security-group-ids option" do
+      before do
+        allow(Fog::Compute::AWS).to receive(:new).and_return(@ec2_connection)
+      end
+
+      context "when invalid input is given for --security-group-ids from cli" do
+        it "raises error" do
+          @knife_ec2_create.config[:security_group_ids] = 'sg-aabbccdd; sg-3764sdss ; sg-00aa11bb'
+          expect { @knife_ec2_create.validate! }.to raise_error(SystemExit)
+        end
+      end
+
+      context "when valid input is given --security-group-ids from cli" do
+        it "does not raise error" do
+          @knife_ec2_create.config[:security_group_ids] = 'sg-aabbccdd, sg-3764sdss, sg-00aa11bb'
+          expect { @knife_ec2_create.validate! }.to_not raise_error
+        end
+      end
+
+      context "when invalid input is given for --security-group-ids from knife.rb" do
```

Comment:
I understand that from the CLI, I wouldn't add array support for the CLI. But using an array would be pretty natural from inside the `knife.rb`.